src/Data/Nat/Primality.agda

```diff
@@ -10,14 +10,29 @@ module Data.Nat.Primality where
 
 open import Data.Empty using (⊥)
 open import Data.Fin.Base using (Fin; toℕ)
-open import Data.Fin.Properties using (all?)
-open import Data.Nat.Base using (ℕ; suc; _+_)
-open import Data.Nat.Divisibility using (_∤_; _∣?_)
-open import Relation.Nullary using (yes; no)
+open import Data.Fin.Properties using (all?; any?)
+open import Data.Nat.Base using (ℕ; suc; _+_; _≤_; z≤n; s≤s)
+open import Data.Nat.Divisibility using (_∣_; _∤_; _∣?_)
+open import Data.Product using (Σ-syntax; _,_)
+open import Relation.Nullary using (yes; no; ¬_)
 open import Relation.Nullary.Decidable using (from-yes)
-open import Relation.Nullary.Negation using (¬?)
+open import Relation.Nullary.Negation using (¬?; decidable-stable)
 open import Relation.Unary using (Decidable)
 
+-- Definition of compositeness
+
+Composite : ℕ → Set
+Composite 0 = ⊥
+Composite 1 = ⊥
+Composite (suc (suc n)) = Σ[ i ∈ Fin n ] 2 + toℕ i ∣ 2 + n
+
+-- Decision procedure for compositeness
+
+composite? : Decidable Composite
+composite? 0 = no λ()
+composite? 1 = no λ()
+composite? (suc (suc n)) = any? λ _ → _ ∣? _
```

Comment:
Minor point, but can you bracket the lambda? I know it's not necessary, but it greatly helps reading it!

Reply:
Sure thing! I agree, it does make it a but more readable